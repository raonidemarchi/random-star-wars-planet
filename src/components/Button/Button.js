import React from 'react';
import styles from './Button.module.scss';

function Button({ children, onClick, disabled }) {
  return (
    <div role="button" className={[styles.button, disabled && styles.isLoading].join(' ')} onClick={disabled ? () => {} : onClick}>
      {children}
    </div>
  )
}

export default Button;
