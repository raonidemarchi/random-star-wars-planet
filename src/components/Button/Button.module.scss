.button {
  padding: 16px 24px;
  min-width: 56px;
  background-color: #f9a825;
  text-transform: uppercase;
  border-radius: 2px;
  box-shadow: 0 10px 20px rgba(0, 0, 0, .2);
  text-align: center;
  letter-spacing: 0.04rem;
  font-weight: 700;
  cursor: pointer;
  user-select: none;

  &.isLoading {
    opacity: .42;
    cursor: initial;
    box-shadow: none;
  }
}