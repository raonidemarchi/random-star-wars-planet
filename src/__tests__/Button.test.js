import React from 'react';
import Button from '../components/Button/Button';
import TestRenderer from 'react-test-renderer';

describe('Button component', () => {
  test('Should render a hello button', () => {
    const tree = TestRenderer.create(
      <Button disabled={false} onClick={() => 'hello'}>
        Hello
      </Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should execute onClick function', () => {
    const message = 'I should the value returned from onClick';
    const tree = TestRenderer.create(
      <Button onClick={() => message} />
    ).toJSON();

    expect(tree.props.onClick()).toBe(message);
  });

  test('Should not trigger on click function', () => {
    const tree = TestRenderer.create(
      <Button disabled={true} onClick={() => `this text shouldn't be returned`}>
        Disabled
      </Button>
    ).toJSON();

    expect(typeof tree.props.onClick()).toBe('undefined');
  });
});