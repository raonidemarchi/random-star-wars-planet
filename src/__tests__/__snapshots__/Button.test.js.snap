// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render a hello button 1`] = `
<div
  className="button false"
  onClick={[Function]}
  role="button"
>
  Hello
</div>
`;
